be/src/storage/lake/lake_persistent_index.cpp

```diff
@@ -329,4 +344,103 @@ void LakePersistentIndex::commit(MetaFileBuilder* builder) {
     builder->finalize_sstable_meta(sstable_meta);
 }
 
+Status LakePersistentIndex::load_from_lake_tablet(TabletManager* tablet_mgr, const TabletMetadataPtr& metadata,
+                                                  int64_t base_version, const MetaFileBuilder* builder) {
+    // 1. create and set key column schema
+    std::shared_ptr<TabletSchema> tablet_schema = std::make_shared<TabletSchema>(metadata->schema());
+    vector<ColumnId> pk_columns(tablet_schema->num_key_columns());
+    for (auto i = 0; i < tablet_schema->num_key_columns(); i++) {
+        pk_columns[i] = (ColumnId)i;
+    }
+    auto pkey_schema = ChunkHelper::convert_schema(tablet_schema, pk_columns);
+
+    // Init PersistentIndex
+    _key_size = PrimaryKeyEncoder::get_encoded_fixed_size(pkey_schema);
+
+    const auto& sstables = metadata->sstable_meta().sstables();
+    int64_t max_sstable_version = sstables.empty() ? 0 : sstables.rbegin()->version();
+    if (max_sstable_version > base_version) {
+        return Status::OK();
+    }
+
+    OlapReaderStatistics stats;
+    std::unique_ptr<Column> pk_column;
+    if (pk_columns.size() > 1) {
+        // more than one key column
+        if (!PrimaryKeyEncoder::create_column(pkey_schema, &pk_column).ok()) {
+            CHECK(false) << "create column for primary key encoder failed";
+        }
+    }
+    vector<uint32_t> rowids;
+    rowids.reserve(4096);
+    auto chunk_shared_ptr = ChunkHelper::new_chunk(pkey_schema, 4096);
+    auto chunk = chunk_shared_ptr.get();
+    auto rowsets = Rowset::get_rowsets(tablet_mgr, metadata);
+    // Rowset whose version is between max_sstable_version and base_version should be recovered.
+    for (auto& rowset : rowsets) {
+        // If it is upgraded from old version of sr, the rowset version will be not set.
+        // The generated rowset version will be treated as base_version.
+        int64_t rowset_version = rowset->version() != 0 ? rowset->version() : base_version;
+        if (rowset_version < max_sstable_version) {
```

Comment:
Why not `if (rowset_version <= max_sstable_version)`, I know the reason, but you need to add some comments about this.

Reply:
added.